great_expectations/rule_based_profiler/parameter_builder/metric_multi_batch_parameter_builder.py

```diff
@@ -64,22 +67,35 @@ def _build_parameters(
         parameters: Optional[Dict[str, ParameterContainer]] = None,
     ):
         """
-        Builds ParameterContainer object that holds ParameterNode objects with attribute name-value pairs and optional details.
-            Args:
-        :return: a ParameterContainer object that holds ParameterNode objects with attribute name-value pairs and optional details
+        Builds ParameterContainer object that holds ParameterNode objects with attribute name-value pairs and optional
+        details.
+
+        :return: ParameterContainer object that holds ParameterNode objects with attribute name-value pairs and
+        ptional details
         """
         validator: Validator = self.get_validator(
             domain=domain,
             variables=variables,
             parameters=parameters,
         )
 
-        batch_id: str = self.get_batch_id(variables=variables)
+        batch_ids: Optional[List[str]] = self.get_batch_ids(
+            domain=domain,
+            variables=variables,
+            parameters=parameters,
+        )
+        if not batch_ids:
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"Utilizing a {self.__class__.__name__} requires a non-empty list of batch identifiers."
+            )
 
         metric_computation_result: Dict[
-            str, Union[Any, Number, Dict[str, Any]]
-        ] = self.get_metric(
-            batch_id=batch_id,
+            str,
+            Union[
+                Union[Any, Number, np.ndarray, List[Union[Any, Number]]], Dict[str, Any]
```

Comment:
`Dict[str, Union[Union[Any ...],Dict[str, Any]]]` is unreasonable (even if we're just making the prior state more explicit).

If metric_computation_result is simply a report object, I'd like to replace it with a dataclass. If it is a rich type we hope to reason about later on, we should promote it to a Domain object and bundle the associated behavior

Reply:
@cdkini @donaldheppner I would prefer to make a metric computation object across all parameters that compute metrics as part of (or after) [JIRA: GREAT-492] -- we need to first solidify the architecture of parameter builders that support metric computations returning results (metric values) of different dimensions.  Moreover, we will also want to consider doing [IPV-250] as part of this as it could help stabilize the output type of metric computations.

Reply:
@cdkini @donaldheppner I added typed objects (`dataclasses`) for the time being, until we implement a more general metric execution state class per my previous comment.  Thanks!